File: frontend/common/urls.js
const serverUrl = window.location.host.includes('localhost')
  ? 'http://localhost:8000'
  : `${window.location.protocol}//${window.location.host}`

const API_URL = `${serverUrl}/api/v1/`

export const MeAPI = `${API_URL}me`
export const LogoutURL = `${API_URL}logout`
export const LoginURL = `${API_URL}login`
export const RegisterURL = `${API_URL}register`

export const REFRESH_TOKEN_URL = `${API_URL}token/refresh`


export const PROJECT_URL = `${API_URL}projects`
export const STAFF_URL = `${API_URL}staffs`

export const PRODUCTS_URL = `${API_URL}products`
export const CONTACT_URL = `${API_URL}contacts`
export const GROUP_URL = `${API_URL}contactgroups`
export const PACKAGES_URL = `${API_URL}packages`
export const CAMPAIGNS_URL = `${API_URL}campaigns`
export const CONTACT_MARKETING_URL = `${API_URL}contact-marketings`

export const NOTES_URL = `${API_URL}notes`
export const MARKETING_PLANS_URL = `${API_URL}marketing-plans`
export const MARKETING_PLANS_CONDITIONS_URL = `${API_URL}marketing-plans-conditions`
export const CONTACTS_MATCH_CONDITIONS_URL = `${API_URL}contacts-match-conditions`
export const FOLLOW_UP_PLANS_URL = `${API_URL}follow-up-plans`
export const PRODUCT_CATEGORIES_URL = `${API_URL}product-categories`
export const PRODUCT_TYPES_URL = `${API_URL}product-types`
export const GET_ACTIONS_URL = `${API_URL}actions`
export const EVENTS_URL = `${API_URL}events`
export const GET_SALE_REPS_URL = `${API_URL}sale-reps`

export const GMAIL_AUTH_URL = `${API_URL}gmail\\auth`
export const GMAIL_SEND_URL = `${API_URL}gmail\\send`

export const WS_NOFICATION_URL = 'ws://localhost:8000/ws/notifications'

export { serverUrl, API_URL }

File: KLTN/urls.py
from django.contrib import admin
from django.urls import include, path
from rest_framework.routers import DefaultRouter
from rest_framework_simplejwt.views import (
    TokenObtainPairView,
    TokenRefreshView,
)

from account.views import MeView, LoginAndUpdateView, LogoutView, RegisterView, ActivateView, SaleRepView, GmailView, GmailExchangeCodeView, SendMessageView
from packages.views import ProductViewSet, PackageViewSet, ProductCategoryView, ProductTypeView
from contacts.views import ContactView, ContactGroupView
from campaigns.views import MarketingPlanView, CampaignView, FollowUpPlanView, GetPlanAction, GetMarketingPlanConditions, ContactMatchConditions, NoteView, CampaignExtraView
from campaigns.views import ContactMarketingView, MailTemplateView
from orders.views import OrderHistoryView, OrderView
from steps.views import StepView, StepDetailView
from events.views import EventView
from reports.views import ReportView

router = DefaultRouter(trailing_slash=False)
# package app
router.register('products', ProductViewSet, base_name='products')
router.register('packages', PackageViewSet, base_name='packages')
router.register('product-categories', ProductCategoryView,
                base_name='product-categories')
router.register('product-types', ProductTypeView, base_name='product-types')
# campaign app
router.register('marketing-plans', MarketingPlanView,
                base_name='marketing_plans')
router.register('follow-up-plans', FollowUpPlanView,
                base_name='follow-up-plans')
router.register('campaigns', CampaignView, base_name='campaigns')
router.register('notes', NoteView, base_name='notes')
router.register('contact-marketings', ContactMarketingView,
                base_name='contact-marketing')
router.register('mail-templates', MailTemplateView,
                base_name='mail-templates')
# order app
router.register('orders', OrderView, base_name='orders')
router.register('order-histories', OrderHistoryView,
                base_name='order-histories')
# contact app
router.register('contacts', ContactView, base_name='contacts')
router.register('contactgroups', ContactGroupView, base_name='contactgroups')
# step app
router.register('steps', StepView, base_name='steps')
router.register('step-detail', StepDetailView, base_name='step-detail')
# events app
router.register('events', EventView, base_name='events')
# reports app
router.register('reports', ReportView, base_name='reports')
# account app
router.register('sale-reps', SaleRepView, base_name='sale-reps')
#------------------------------------------------------------------------------------#

urlpatterns = []
urlpatterns += router.urls

urlpatterns = [
    path('jet/', include('jet.urls', 'jet')),
    path('admin/', admin.site.urls),
    path('api/v1/', include(router.urls)),
    path('api/v1/token', TokenObtainPairView.as_view(),
         name='token_obtain_pair'),
    path('api/v1/token/refresh',
         TokenRefreshView.as_view(), name='token_refresh'),

    # account views
    path('api/v1/me', MeView.as_view()),
    path('api/v1/login', LoginAndUpdateView.as_view()),
    path('api/v1/login/<int:pk>', LoginAndUpdateView.as_view()),
    path('api/v1/logout', LogoutView),
    path('api/v1/register', RegisterView.as_view()),
    path('api/v1/activate', ActivateView),
    path('api/v1/actions', GetPlanAction),
    path('api/v1/marketing-plans-conditions', GetMarketingPlanConditions),
    path('api/v1/contacts-match-conditions', ContactMatchConditions),

    path('api/v1/campaigns/<int:pk>/note',
         CampaignExtraView.as_view({'get': 'note'})),

    path('api/v1/gmail/auth', GmailView),
    path('api/v1/gmail/exchange', GmailExchangeCodeView),
    path('api/v1/gmail/send', SendMessageView.as_view()),
]
